pyproject.toml

```
@@ -29,10 +29,16 @@ requires = [
     "numpy==1.20.0; python_version=='3.8' and platform_machine=='arm64' and platform_system=='Darwin'",
     "numpy==1.20.0; python_version=='3.9' and platform_machine=='arm64' and platform_system=='Darwin'",
 
+    # loongarch64 requires numpy>=1.22.0
+    "numpy==1.22.0; platform_machine=='loongarch64'",
+
     # default numpy requirements
-    "numpy==1.18.5; python_version=='3.8' and (platform_machine!='arm64' or platform_system!='Darwin') and platform_machine!='aarch64' and platform_python_implementation != 'PyPy'",
-    "numpy==1.19.3; python_version=='3.9' and (platform_machine!='arm64' or platform_system!='Darwin') and platform_python_implementation != 'PyPy'",
-    "numpy==1.21.4; python_version=='3.10' and platform_python_implementation != 'PyPy'",
+    "numpy==1.18.5; python_version=='3.8' and (platform_machine!='arm64' or platform_system!='Darwin') and platform_machine!='aarch64' and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'",
+    "numpy==1.19.3; python_version=='3.9' and (platform_machine!='arm64' or platform_system!='Darwin') and platform_machine!='loongarch64' and platform_python_implementation != 'PyPy'",
```

I understand that this contains the negation of the Darwin+arm64 from above, but since aarch64 and arm64 are distinguished, arm64 does not seem possible with anything _but_ Darwin**? Could this then not be simplified as follows?

```suggestion
    "numpy==1.18.5; python_version=='3.8' and platform_machine not in ('arm64', 'aarch64', 'loongarch64') and platform_python_implementation != 'PyPy'",
    "numpy==1.19.3; python_version=='3.9' and platform_machine not in ('arm64', 'loongarch64') and platform_python_implementation != 'PyPy'",
```

I find this quite a bit easier to parse.

** In other words, the only semantic change of my suggestion is that it effectively does not account for arm64+non-Darwin, but I'm not sure that exists?

---

OK, github formatting is weird on this suggestion. Here's the raw code:
```
    "numpy==1.18.5; python_version=='3.8' and platform_machine not in ('arm64', 'aarch64', 'loongarch64') and platform_python_implementation != 'PyPy'",
    "numpy==1.19.3; python_version=='3.9' and platform_machine not in ('arm64', 'loongarch64') and platform_python_implementation != 'PyPy'",
```